pkg/mutator/http_handler_factory.go

```diff
@@ -0,0 +1,183 @@
+package mutator
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-admission-controller/pkg/filter"
+	"github.com/giantswarm/azure-admission-controller/pkg/generic"
+)
+
+type HttpHandlerFactoryConfig struct {
+	CtrlClient client.Client
+}
+
+// HttpHandlerFactory creates HTTP handlers for mutating create and update requests.
+type HttpHandlerFactory struct {
+	ctrlClient client.Client
+}
+
+func NewHttpHandlerFactory(config HttpHandlerFactoryConfig) (*HttpHandlerFactory, error) {
+	if config.CtrlClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CtrlClient must not be empty", config)
+	}
+
+	h := &HttpHandlerFactory{
+		ctrlClient: config.CtrlClient,
+	}
+
+	return h, nil
+}
+
+// NewCreateHandler returns a HTTP handler for mutating create requests.
+func (h *HttpHandlerFactory) NewCreateHandler(mutator WebhookCreateHandler) http.HandlerFunc {
+	mutateFunc := func(ctx context.Context, review v1beta1.AdmissionReview) ([]PatchOperation, error) {
+		// Decode the new CR from the request.
+		object, err := mutator.Decode(review.Request.Object)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		ownerClusterGetter := func(objectMeta metav1.ObjectMetaAccessor) (capi.Cluster, bool, error) {
+			ownerCluster, ok, err := generic.GetOwnerCluster(ctx, h.ctrlClient, object)
+			if err != nil {
+				return capi.Cluster{}, false, microerror.Mask(err)
+			}
+
+			return ownerCluster, ok, nil
+		}
+
+		// Check if the CR should be mutated by the azure-admission-controller.
+		ok, err := filter.IsObjectReconciledByLegacyRelease(ctx, h.ctrlClient, object, ownerClusterGetter)
```

Comment:
Hmm this is incorrect, cache should be passed here. Same in few other places. I'll fix that.